Can't debug_sql! query that contains time::Timespec; trait not implemented

## Setup

### Versions

- **Rust:** stable 1.18.0
- **Diesel:** 0.13.0
- **Database:** postgres
- **Operating System** osx

### Feature Flags

- **diesel:** features = ["postgres", "serde_json", "deprecated-time", "chrono"]
- **diesel_codegen:** none

## Problem Description


### What are you trying to accomplish?

I'm trying to debug a query in crates.io (essentially [this code](https://github.com/rust-lang/crates.io/blob/f94a74d7b0cc7ebd028cdcebeab45abc46d7b322/src/version.rs#L464-L469) but i'm changing some things for debugging), because i'm having a timezone issue:

```
    let downloads_query = VersionDownload::belonging_to(&version)
        .filter(version_downloads::date.between(
            date(cutoff_start_date)..
                date(cutoff_end_date),
        ))
        .order(version_downloads::date);

    debug_sql!(downloads_query);
```

### What is the expected output?

I expect to see a printed version of the SQL generated by this diesel code.


### What is the actual output?

I get a compilation error:

```
error[E0277]: the trait bound `time::Timespec: diesel::types::ToSql<diesel::types::Timestamp, diesel::backend::Debug>` is not satisfied
   --> src/version.rs:475:5
    |
475 |     debug_sql!(downloads_query);
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `diesel::types::ToSql<diesel::types::Timestamp, diesel::backend::Debug>` is not implemented for `time::Timespec`
    |
```

### Steps to reproduce

Check out this branch of crates.io: https://github.com/integer32llc/crates.io/tree/diesel-debug-sql-error and try to build it.

## Checklist

- [x] I have already looked over the [issue tracker](https://github.com/diesel-rs/diesel/issues) for similar issues.

--- a/diesel/src/types/impls/debug.rs
+++ b/diesel/src/types/impls/debug.rs
@@ -30,6 +30,15 @@ mod pg_impls {
     debug_to_sql!(Interval, PgInterval);
 }
 
+#[cfg(feature = "deprecated-time")]
+mod deprecated_time_impls {
+    extern crate time;
+    use super::*;
+    use self::time::Timespec;
+
+    debug_to_sql!(Timestamp, Timespec);
+}
+
 #[cfg(feature = "chrono")]
 mod chrono_impls {
     extern crate chrono;

